doc/book/app_server/luajit_memprof.rst

```diff
@@ -0,0 +1,527 @@
+
+LuaJIT memory profiler
+======================
+
+Starting from version :doc:`2.7.1 </release/2.7.1>`, Tarantool
+has the built‑in module called ``misc.memprof`` that implements the LuaJIT memory
+profiler (further, *profiler*). The profiler provides
+a memory allocation report that helps analyse Lua code and find out the places
+that put the most pressure on the Lua garbage collector (GC).
+
+..  contents::
+    :local:
+    :depth: 2
+
+.. _profiler_usage:
+
+Working with profiler
+---------------------
+
+Usage of the profiler is two-fold:
+
+1.  :ref:`Collect <profiler_usage_get>` a binary profile of allocations,
+    reallocations, and deallocations in memory related to Lua VM
+    (further, *binary memory profile* or *binary profile* for short).
+2.  :ref:`Parse <profiler_usage_parse>` the binary profile collected and get
+    a human-readable profiling report.
+
+.. _profiler_usage_get:
+
+Collecting binary profile
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To collect a binary profile for a particular part of the Lua code,
+you need to place this part between two ``misc.memprof`` functions,
+namely, ``misc.memprof.start()`` and ``misc.memprof.stop()``, and then execute
+the code under Tarantool.
+
+Below is a chunk of simple Lua code named ``test.lua`` to illustrate this.
+
+.. _profiler_usage_example01:
+
+..  code-block:: lua
+    :linenos:
+
+    -- Prevent allocations on traces.
+    jit.off()
+    local str, err = misc.memprof.start("memprof_new.bin")
+    -- Lua doesn't create a new frame to call string.rep, and all allocations
+    -- are attributed not to the append() function but to the parent scope.
+    local function append(str, rep)
+        return string.rep(str, rep)
+    end
+
+    local t = {}
+    for i = 1, 1e5 do
+        -- table.insert is the built‑in function and all corresponding
```

Comment:
@veod32, sorry for confusing you a bit. Talking about non-breaking hyphen, I meant only those used in the plain text. Since this code can be copied and pasted as is, it's better to leave everything in ASCII in the chunk (i.e. use the plain hyphen) to avoid possible lexer crash.

Reply:
Ok, I see. Corrected.